tools/cli/main.c

```diff
@@ -63,13 +69,26 @@ int main(int argc, char* argv[])
     PTDNF pTdnf = NULL;
     int nFound = 0;
 
-    //granular permissions for non root users are pending.
-    //blocking all operations for non root and show the
-    //right error to avoid confusion.
-    if(geteuid() != 0)
+    /*
+     * granular permissions for non root users are pending.
+     * blocking all operations for non root and show the
+     * right error to avoid confusion
+     */
+    if (geteuid())
     {
-        dwError = ERROR_TDNF_PERM;
-        BAIL_ON_CLI_ERROR(dwError);
+        BAIL_ON_CLI_ERROR((dwError = ERROR_TDNF_PERM));
+    }
+
+    if (IsTdnfAlreadyRunning())
```

Comment:
Again, only modify operations needs to take this lock. We must defer this call further down, maybe at per-comand bassis.